Source/WebCore/dom/ImageOverlay.cpp

```diff
@@ -490,7 +480,35 @@ void updateWithTextRecognitionResult(HTMLElement& element, const TextRecognition
         return quad;
     };
 
-    bool applyUserSelectAll = document->isImageDocument() || renderer->style().userSelect() != UserSelect::None;
+    bool smallImageWithSingleWord = [&] {
+        constexpr auto smallImageDimensionThreshold = 64;
+        if (containerRect.width() > smallImageDimensionThreshold || containerRect.height() > smallImageDimensionThreshold)
+            return false;
+
+        return result.blocks.isEmpty() && result.lines.size() == 1 && result.lines[0].children.size() == 1;
+    }();
+
+    if (smallImageWithSingleWord)
+        return;
+
+    auto elements = updateSubtree(element, result);
+    if (!elements.root)
+        return;
+
+    {
```

Comment:
why is this separate scope needed?

Reply:
Oh, I wrote it this way to ensure that nothing below tries to reuse `renderer` here. If anything updates layout below and causes the image to recreate its renderer, then this pointer could point to a destroyed object.